kernel/xpmem_misc.c

```diff
@@ -315,21 +320,148 @@ xpmem_debug_printk_procfs_open(struct inode *inode, struct file *file)
 	return single_open(file, xpmem_debug_printk_procfs_show, NULL);
 }
 
+static long
+user_atol(const char __user *buffer, size_t count)
```

Comment:
isn't there kernel function for this conversion?

Reply:
now using kstrol() available at least from 3.10